client/src/components/QuestionsAnswers/QuestionsList.jsx

```diff
@@ -60,27 +73,75 @@ const QuestionsList = ({ productId }) => {
   useEffect(() => {
     getProductName();
   }, []);
+  useEffect(() => {
+    if (searchValue.length) {
+      if (searchValue.length > 2) {
+        const allQuestions = questions.filter(question => (
+          question.question_body.includes(searchValue)
+        ));
+        setFilteredQ(allQuestions);
+        setIsFiltered(true);
+      } else {
+        setFilteredQ([]);
+        setIsFiltered(false);
+      }
+    } else {
+      setFilteredQ([]);
+      setIsFiltered(false);
+    }
+  }, [searchValue]);
 
   return (
-    <div>
-      { questions.length
-        ? questions.slice(0, count).map(question => (
-          <Question
-            key={ question.question_id }
-            question={ question }
-            productName={ product } />
-        ))
-        : null
+    <Container className='questions-list-container'>
+      <Container className='questions-container'>
+        { isFiltered && filteredQ.length
+          ? filteredQ.slice(0, count).map(question => (
+            <Question
+              key={ question.question_id }
+              question={ question }
+              productName={ product }
+              answers={ question.answers } />
+          ))
+          : isFiltered && !filteredQ.length
+            ? <div>No questions with that search criteria can be found.</div>
+            : questions.length && !filteredQ.length && !isFiltered
+              ? questions.slice(0, count).map(question => (
+                <Question
+                  key={ question.question_id }
+                  question={ question }
+                  productName={ product }
+                  answers={ question.answers } />
+              ))
+              : null
+        }
+      </Container>
+      { isFiltered
+        ? filteredQ.length
+          ? count < filteredQ.length && filteredQ.length > 2
+            ? <Button className='toggle-show-questions' onClick={ showMoreQuestions }>More Answered Questions</Button>
+            : null
+          : null
+        : questions.length > 2 && remainingQ
+          ? <Button className='toggle-show-questions' onClick={ showMoreQuestions }>More Answered Questions</Button>
+          : null
       }
-      { questions.length > 2 && remainingQ
-        ? <button onClick={ showMoreQuestions }>More Answered Questions</button>
+      {' '}
```

Comment:
What is this line doing?

Reply:
The `{' '}` line just adds a break/space between the two Buttons inside my Modal. Without this line, the buttons are right next to each other with no space in between.

Reply:
You can't do that with a margin?